engine/src/org/pentaho/di/trans/steps/memgroupby/MemoryGroupBy.java

```diff
@@ -75,16 +75,13 @@ public MemoryGroupBy( StepMeta stepMeta, StepDataInterface stepDataInterface, in
     data = (MemoryGroupByData) stepDataInterface;
   }
 
+  @Override
   public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws KettleException {
     meta = (MemoryGroupByMeta) smi;
     data = (MemoryGroupByData) sdi;
 
     Object[] r = getRow(); // get row!
     if ( first ) {
-      if ( r == null ) {
```

Comment:
Shouldn't the change only happen if meta.isAlwaysGivingBackOneRow() is true ? Seems to me this needs to go back as:

if ( ( r == null ) && ( !meta.isAlwaysGivingBackOneRow() ) ) { ...